pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -168,7 +168,7 @@ func (r *ReconcileMongoDB) Reconcile(request reconcile.Request) (reconcile.Resul
 	}
 
 	if instance.Spec.MongoDBUser == "" {
-		user = "admin"
+		user = createRandomAlphaNumeric(8)
```

Comment:
If `MongoDBUser` and `MongoDBPass` are required.  Do we still need to check`if instance.Spec.MongoDBUser == ""` and `if instance.Spec.MongoDBPass == "" ` ?

Reply:
yes, if the user fails to provide them then they will be given a warning along with a way to get around the warning. They could then re do the command with `--validate=false` and it will not give them the warning again. In this scenario, we should use random strings.